File: 백준/Silver/23516. Least Common Divisor/Least Common Divisor.py
S,T=input(),input()
def R(s,U):return s==U*(len(s)//len(U))
def G(a, b):
 while b:a,b=b,a%b
 return a
E=G(len(S),len(T))
for i in range(1,E+1):
 if E%i==0:
  P=S[:i]
  if R(S,P) and R(T,P):
   print(P)
   break
else:print("No solution")

File: 백준/Silver/23516. Least Common Divisor/Least Common Divisor.c
G(int a,int b){return b?G(b,a%b):a;}
R(char *s,char *P,int T){int Z=strlen(s);if(Z%T!=0)return 0;for(int i=0;i<Z;i++){if(s[i]!=P[i%T])return 0;}return 1;}
main(X,Y,e,i){char S[51],T[51];scanf("%s%s",S,T);X=strlen(S);Y=strlen(T);e=G(X, Y);for (i=1;i<=e;i++){if(e%i==0&&R(S,S,i)&&R(T,S,i)){S[i]='\0';printf("%s",S);return 0;}}printf("No solution");}